pyne/alara.py

```diff
@@ -890,5 +892,242 @@ def _find_phsrc_dc(idc, phtn_src_dc):
+        s += "flux flux_{0} {1} {2} 0 default\n".format(i, fluxin_file, flux_magnitude)
+    s += "output zone\n"
+    s += "integrate_energy\n"
+    # Photon energy bin structure
+    # 24 bin structure
+    if num_p_groups == 24:
+        s += "    photon_source {0}/fendl2.0bin {1} 24 1.00E4 2.00E4 5.00E4 1.00E5\n".format(data_dir, phtn_src_file)
+        s += "    2.00E5 3.00E5 4.00E5 6.00E5 8.00E5 1.00E6 1.22E6 1.44E6 1.66E6\n"
+        s += "    2.00E6 2.50E6 3.00E6 4.00E6 5.00E6 6.50E6 8.00E6 1.00E7 1.20E7\n"
+        s += "    1.40E7 2.00E7\nend\n"
+    # 42 bin structure
+    elif num_p_groups == 42:
+        s += "    photon_source {0}/fendl2.0bin {1} 42\n".format(data_dir, phtn_src_file)
+        s += "     1e4 2e4 3e4 4.5e4 6e4 7e4 7.5e4 1e5 1.5e5 2e5 3e5 4e5\n"
+        s += "     4.5e5 5.1e5 5.12e5 6e5 7e5 8e5 1e6 1.33e6 1.34e6 1.5e6 1.66e6 2e6\n"
+        s += "     2.5e6 3e6 3.5e6 4e6 4.5e6 5e6 5.5e6 6e6 6.5e6 7e6 7.5e6 8e6 1e7\n"
+        s += "     1.2e7 1.4e7 2e7 3e7 5e7\nend\n"
+    s += "pulsehistory my_schedule\n"
+    s += "    1 0.0 s\nend\n"
+    s += "schedule total\n"
+    for i, irr_time in enumerate(irr_times):
+        s += "    {0} s flux_{1} my_schedule 0 s\n".format(irr_time, i)
+    s += "end\n"
+    s += "cooling\n"
+    for d in decay_times:
+        s += "    {0} s\n".format(d)
+    s += "end\n"
+    with open(input_file, 'w') as f:
+        f.write(s)
+
+def _gt_alara(data_dir, mats, neutron_spectrum, flux_magnitudes, irr_times, 
+              decay_times, num_p_groups, run_dir):
+    """
+    Function that prepares necessary input files and runs ALARA
+    
+    Parameters
+    ----------
+    data_dir : str
+        Path to nuclib file
+    mats : list of Material objects
+        List of properties of materials in the geometry
+    neutron_spectrum : list
+        Neutron spectrum (length is equal to number of n energy groups)
+    flux_magnitudes : list
+        Magnitude of flux in each neutron energy group
+    irr_times : list
+        Irradiation times [s]
+    decay_times : list
+        Decay times [s]
+    num_p_groups : int
+        Number of photon energy groups for ALARA calculation
+    run_dir : str
+        Directory to store ALARA run input and output files
 
+    Returns
+    ----------
+    phtn_src_file : str
+        Path to photon source file produced by ALARA
+    """
+    neutron_spectrum = _gt_normalize(neutron_spectrum)
+    num_n_groups = len(neutron_spectrum)
+    num_mats = len(mats)
+    num_decay_times = len(decay_times)
+    num_irr_times = len(irr_times)
+
+    # Write matlib file
+    matlib_file = os.path.join(run_dir, "matlib")
+    _gt_write_matlib(mats, matlib_file)
+
+    # Write fluxin file
+    fluxin_file = os.path.join(run_dir, "fluxin")
+    fluxes = []
+    for m in range(num_mats):
+        for n in range(num_n_groups):
+            fluxes.append([neutron_spectrum[n] if x ==
+                           n else 0 for x in range(num_n_groups)])
```

Comment:
This appears to do an extra loop that accomplishes nothing.

Reply:
In ALARA fluxin file we will have 175 flux blocks per material (one block per energy group). In each block all groups are zero except for one group. This way we can see that the "for n in .. " is equivalent to looping over the block index while in "flux.append .... for x in ..." we loop over the energy group index within that block and fill only one group with a value and the rest with zero.
An additional outer loop will be added later when we include the option of many decay times.